### DISCUSSION TITLE
Too many line breaks when using :: when format scala

### DISCUSSION THREAD

**Discussion by Zouxxyy** [diffplug/spotless#1982] [Q&A]
Hi, team, we use mvn + spotless + scalafmt to format our scala code, I really like its automatic reformat function, now we meet a problem, hope to get help

## Configuration (required) ##

```
spotless.version=2.13.0
.scalafmt.conf version = 3.4.3
```

full conf, see https://github.com/apache/incubator-paimon/blob/master/.scalafmt.conf

## Command-line parameters (required) ##

```shell
mvn spotless:apply
```

## Steps

Given code like this:
```scala
val expertResult2 =
    Row("+I", 1, "v_1") :: Row("-U", 1, "v_1") :: Row("+U", 1, "v_1_new") :: Row("+I", 2, "v_2") :: 
       Row("-U", 2, "v_2") :: Row("+U", 2, "v_2_new") :: Row("+I", 3, "v_3") :: Nil
```

## Problem

Scalafmt formats code like this:
```scala
val expertResult2 =
    Row("+I", 1, "v_1") :: Row("-U", 1, "v_1") :: Row("+U", 1, "v_1_new") :: Row(
        "+I",
         2,
         "v_2") :: Row("-U", 2, "v_2") :: Row("+U", 2, "v_2_new") :: Row("+I", 3, "v_3") :: Nil
```

## Expectation

The code after format is a bit strange, don't want see too many line breaks, is there any relevant configuration that can improve it, thanks

**Comment by nedtwigg** [comment]
This is likely a bug in `scalafmt`, not `spotless`. If `scalafmt` fixes it, you can easily set spotless to use this one.
